boards/wio_terminal/src/sensors.rs

```diff
@@ -47,24 +55,28 @@ impl Accelerometer {
 }
 
 /// Analog Light Sensor
-pub struct LightSensor {
+pub struct LightSensor<P>
+where
+    P: Into<LightSensorAdc>,
+{
     /// Analog Light Sensor input pin
-    pub pd1: Pd1<Input<Floating>>,
+    pub pd1: P,
 }
```

Comment:
Same comment as above. I think the struct should hold a concrete type, not a generic type. The `init` function should be generic.